go/ngql/types.go

```diff
@@ -419,32 +419,41 @@ func getListElements(vrw types.ValueReadWriter, v types.Value, args map[string]i
 	}
 
 	// Clamp ranges
-	if count <= 0 || idx >= len {
+	if count <= 0 || idx >= end {
 		return ([]interface{})(nil)
 	}
 	if idx < 0 {
 		idx = 0
 	}
-	if idx+count > len {
-		count = len - idx
+	if idx+count > end {
+		count = end - idx
 	}
 
 	values := make([]interface{}, count)
 
-	// This is a workaround for allowing types.Set as types.List.
-	type listOrSetIterator interface {
-		Next() types.Value
-	}
+	// @raf a few questions:
+	//  - why do we do MaybeGetScalar below?
```

Comment:
This is so that the GraphQL query/request bottoms out. The scalars (bool, float64, string) are the base types that the graphql library uses to know that no more resolving is needed.